zstacklib/zstacklib/utils/fileUploader.py

```
@@ -0,0 +1,289 @@
+            length = min(self.size - self.offset, n)
+            content = self.image.read(self.offset, length)
+            self.offset += length
+            return content
+
+        def write(self, content):
+            self.image.write(content, self.offset)
+            self.offset = min(self.offset + len(content), self.size)
+
+        def close(self):
+            self.image.close()
+            logger.debug("%s closed" % str(self.image))
```

Ruff (0.13.1)

`logger` may be undefined, or defined from star imports

(F405)